### PR TITLE
Expose rounding modes as constructors + add BigNumeric docs.

### PR DESCRIPTION
**Pull request by sofiafaro-da**
This PR exposes the rounding modes as constructors for RoundingMode. Pattern matching for RoundingMode is not implemented and not critical (will open a separate issue).

This PR also adds documentation for BigNumeric.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/daml/reference/data-types.rst [hunk 1]
```diff
@@ -33,7 +33,11 @@ Table of built-in primitive types
    * - ``Numeric n``
      - fixed point decimal numbers
      - ``1.0``
-     - `Numeric n` values are rational numbers with up to ``38`` digits. The scale parameter ``n`` controls the number of digits after the decimal point, so for example, ``Numeric 10`` values have 10 decimal places, and ``Numeric 20`` values have 20 decimal places. The value of ``n`` must be between ``0`` and ``37`` inclusive.
+     - ``Numeric n`` values are rational numbers with up to ``38`` decimal digits. The scale parameter ``n`` controls the number of digits after the decimal point, so for example, ``Numeric 10`` values have 10 decimal places, and ``Numeric 20`` values have 20 decimal places. The value of ``n`` must be between ``0`` and ``37`` inclusive.
```
Comment (cocreature) [inline docs/source/daml/reference/data-types.rst:36]: hm, I view Numeric as always having 38 decimal digits including trailing zeroes.

File: compiler/damlc/daml-prim-src/GHC/Types.daml [hunk 2]
```diff
@@ -208,13 +208,37 @@ data Decimal =
...
+      -- are equidistant, in which case round towards zero.
+  | RoundingHalfEven
+      -- ^ Round towards the nearest neighbor unless both neighbors
+      -- are equidistant, in which case round towards the even neighbor.
+  | RoundingHalfUp
+      -- ^ Round towards the nearest neighbor unless both neighbors
+      -- are equidistant, in which case round away from zero.
+  | RoundingUnnecessary
+      -- ^ Do not round. Raises an error if the result cannot
+      -- be represented without rounding.
```
Comment (cocreature) [inline compiler/damlc/daml-prim-src/GHC/Types.daml:242]:
```suggestion
      -- be represented without rounding at the targeted scale.
```

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: docs/source/daml/reference/data-types.rst (hunk 1 above)
Comment (sofiafaro-da) [reply docs/source/daml/reference/data-types.rst]: How's this? https://github.com/digital-asset/daml/pull/9336/commits/ca9a518c3879ec2f9e22bfcd1cf42debf8061601

Comment (cocreature) [reply docs/source/daml/reference/data-types.rst]: I like it, thanks!

**Comment by remyhaemmerle-da** [comment]
part of #8719

**Pull request merged**